src/basic/fileio.c

```diff
@@ -355,6 +355,21 @@ int write_string_filef(
         return write_string_file(fn, p, flags);
 }
 
+int write_base64_file_at(
+                int dir_fd,
+                const char *fn,
+                const struct iovec *data,
+                WriteStringFileFlags flags) {
+
+        _cleanup_free_ char *encoded = NULL;
+        ssize_t n;
+        n = base64mem_full(data ? data->iov_base : NULL, data ? data->iov_len : 0, 79, &encoded);
```

Comment:
Hmm, I think we should assert on data here. It's really pointless to call this with NULL iovec?

Reply:
Well, we usually treat NULL Sets, strv, and so on as equivalent to empty. I think we should do the same for iovec.

Reply:
(simply to minimize surprises and be systematic here)

Reply:
(I don't like this practice TBH, but please at least add a newline between var declaration and function call?)

Reply:
we didn't use to do this, but we started doing this once Unit objects gained one Set* for each dep type, and it just makes so much sense to treat NULL as an empty set then, because it means zero allocations for sparse sets of deps (which they generally are).

From there it propagated to our other types, because the fact you can avoid all allocations for one of the most common special cases is really worth it.

From my current perspective I must say: best idea ever: it's not just more memory efficient but also a lot shorter in code (since you don't have to explicitly initialize these objects hence). Hence, from my side I only see benefits of this practice.

Reply:
I think it's better to make the function call only if you really have something to do. Then, the callee should check the params.

This way we can detect if the caller wants to do something, but something went wrong and the arg is NULL